wal/live_reader.go

```diff
@@ -27,26 +27,31 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+type LiveReaderMetrics struct {
+	ReaderCorruptionErrors *prometheus.CounterVec
+}
+
+func NewLiveReaderMetrics() *LiveReaderMetrics {
```

Comment:
Should this take a registry and register all metrics as well? It would be easy for a caller to miss registering one in the future.

Reply:
Yes I thought that is the pattern - to pass registry here in constructor.